src/sentry/runner/initializer.py

```diff
@@ -82,44 +82,60 @@ def get_asset_version(settings):
 }
 
 
-def bootstrap_options(settings, config):
+def bootstrap_options(settings, config=None):
     """
     Quickly bootstrap options that come in from a config file
     and convert options into Django settings that are
     required to even initialize the rest of the app.
     """
-    if config is None:
-        return
+    options = {}
+    if config is not None:
+        # Attempt to load our config yaml file
+        from sentry.utils.yaml import safe_load
+        from yaml.parser import ParserError
+        from yaml.scanner import ScannerError
+        try:
+            with open(config, 'rb') as fp:
+                options = safe_load(fp)
+        except IOError:
+            # Gracefully fail if yaml file doesn't exist
+            pass
+        except (AttributeError, ParserError, ScannerError) as e:
+            from .importer import ConfigurationError
+            raise ConfigurationError('Malformed config.yml file: %s' % unicode(e))
+
+        # Empty options file, so fail gracefully
+        if options is None:
+            options = {}
+        # Options needs to be a dict
+        elif not isinstance(options, dict):
+            from .importer import ConfigurationError
+            raise ConfigurationError('Malformed config.yml file')
+
     from sentry.conf.server import DEAD
-    from sentry.utils.yaml import safe_load
-    from yaml.parser import ParserError
-    from yaml.scanner import ScannerError
-    try:
-        with open(config, 'rb') as fp:
-            options = safe_load(fp)
-    except IOError:
-        # Gracefully fail if yaml file doesn't exist
-        return
-    except (AttributeError, ParserError, ScannerError) as e:
-        from .importer import ConfigurationError
-        raise ConfigurationError('Malformed config.yml file: %s' % unicode(e))
-    # Empty options file, so fail gracefully
-    if options is None:
-        return
-    # Options needs to be a dict
-    if not isinstance(options, dict):
-        from .importer import ConfigurationError
-        raise ConfigurationError('Malformed config.yml file')
+
     # First move options from settings into options
     for k, v in options_mapper.iteritems():
-        if hasattr(settings, v) and k not in options and getattr(settings, v) is not DEAD:
+        if getattr(settings, v, DEAD) is not DEAD and k not in options:
+            warnings.warn(
```

Comment:
This check had to be moved in here, since below, these settings get moved from SENTRY_OPTIONS into `settings.*`, which means during `apply_legacy_settings`, the check is invalid. This `DEAD` marker allows to differentiate at this point.